### PR TITLE
Add installation instructions to the docs

### PR DESCRIPTION
**Pull request by SimonBarendse**
Many people are still on Terraform 0.12 and earlier versions, so it should be clear from the registry page how to use the provider for those Terraform versions.

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -15,6 +15,32 @@ You can set environment variable `SECRETHUB_CREDENTIAL` or read it from disk usi
...
+```
+
+### Terraform 0.12 and earlier
+
+For Terraform 0.12 and earlier, you have to download the binary for your operating system and architecture from [GitHub releases](https://github.com/secrethub/terraform-provider-secrethub/releases) yourselves and place it in `~/.terraform.d/plugins` (`%APPDATA%\terraform.d\plugins` on Windows).
+
+For linux amd64, you can do so with the following command:
+```sh
+mkdir -p ~/.terraform.d/plugins && curl -SfL https://github.com/secrethub/terraform-provider-secrethub/releases/latest/download/terraform-provider-secrethub-linux-amd64.tar.gz | tar zxf - -C ~/.terraform.d/plugins
+```
```
Comment (florisvdg) [inline docs/index.md:42]: We could also just link to our nice install tabs here: https://secrethub.io/docs/guides/terraform/#install
So you get macOS, Windows, and 386 too. And we keep it short here, e.g.:

````suggestion
For Terraform 0.12 and earlier, you have to [place the binary in the plugin directory](https://secrethub.io/docs/guides/terraform/#install)
````

File: docs/index.md [hunk 2]
```diff
@@ -15,6 +15,29 @@ You can set environment variable `SECRETHUB_CREDENTIAL` or read it from disk usi
...
+### Terraform 0.13 and later
+
+To install this provider, copy and paste this code into your Terraform configuration. Then, run `terraform init`.
+
+```hcl
+terraform {
+  required_providers {
+    secrethub = {
+      source = "secrethub/secrethub"
+      version = "1.2.2"
```
Comment (jpcoenen) [inline docs/index.md:29]: Do we want to add this static version here? We can also just refer to the `Install Provider` button on the top right of this page, then it's always up-to-date.

Comment (SimonBarendse) [reply docs/index.md:29]: Alternatively, we could auto-bump it like we do a.o. in the pipeline for our GitHub action: https://github.com/secrethub/actions/blob/v0.2.0/.github/workflows/bump-version.yml

Comment (jpcoenen) [reply docs/index.md:29]: Could work, but isn't it just as easy to refer to the `Use Provider` button? It's literally on the same page.

Comment (SimonBarendse) [reply docs/index.md]: I think including it in the page is slightly more user friendly, but I'm okay with referring to the button.

File: .github/workflows/bump-version.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          token: ${{ env.GITHUB_TOKEN }}
+      - name: Bump version
+        uses: florisvdg/action-version-bump@v0.1.0
+        with:
+          sed: 's/\(version = "\).*"/\1$VERSION"/g'
+          file: docs/index.md
+          author_email: bender.github@secrethub.io
```
Comment (jpcoenen) [inline .github/workflows/bump-version.yml]: What's the purpose of this now?

Comment (SimonBarendse) [reply .github/workflows/bump-version.yml]: I've removed this in the next commit: fb630bf

This commit was from before I applied your suggestion of referring to the button.

Comment (jpcoenen) [reply .github/workflows/bump-version.yml]: Ah. Looked over that because the whole file was deleted. :+1: in that case.

**Review by jpcoenen — APPROVED** [review]

**Pull request merged**